### DISCUSSION TITLE
Perl binary lässt sich nicht starten (falscher Interpreter ?)

### DISCUSSION THREAD

**Discussion by gitigitihub** [Freetz-NG/freetz-ng#745] [Allgemein]
Hallo,
ich habe ein Aktuelles Freetz-NG Image gebaut um FHEM zu nutzen.
Leider funktioniert das Perl paket nicht. Das Binary lässt sich nicht starten.
Augabe von Strace  : 
execve("/usr/bin/perl", ["perl"], 0x7fd06020 /* 230 vars */) = -1 ENOENT (No such file or directory)
strace: exec: No such file or directory
+++ exited with 1 +++ 

Beim vergleich des Binary ist mir aufgefallen das der Interpreter eventuell falsch gelinkt ist :
perl: ELF 32-bit MSB executable, MIPS, MIPS32 rel2 version 1 (SYSV), dynamically linked, interpreter /lib/ld-uClibc.so.0, stripped

(/lib/ld-uClibc.so.0, gibt es nicht)

Bei anderen Binarys ist das anders z.b. PHP:
php: ELF 32-bit MSB executable, MIPS, MIPS32 rel2 version 1 (SYSV), dynamically linked, interpreter /usr/lib/freetz/ld-uClibc.so.1, stripped

Ich habe im Perl Paket leider nicht die Stelle gefunden wo dies eingestellt wird.
Hat jemeand eine Idee wie man dies beheben kann ?

Viele Grüße
RK

**Comment by fda77** [comment]
Perl nutzt nicht das "install" von freetz, und so wiurd der interperte nicht angepasst. Es müsste so (unschön) wie bei python funktionieren:
https://github.com/Freetz-NG/freetz-ng/blob/master/make/pkgs/python/python.mk#L111
Das patchelf kann man auch nachträglich zum testen ausführen
